app/services/arcgis_api/geocoder_factory.rb

```diff
@@ -0,0 +1,11 @@
+module ArcgisApi
+  class GeocoderFactory
+    def create
+      if IdentityConfig.store.arcgis_mock_fallback
+        ArcgisApi::Mock::Geocoder.new
+      else
+        ArcgisApi::Geocoder.new
+      end
+    end
+  end
```

Comment:
proposal:
1. call this a Router to match our `DocAuthRouter`
2. make it a module, or at least ditch the instance method, there's nothing really to instantiate here because there's no state?
```suggestion
  module GeocoderRouter
    module_function

    def geocoder
      if IdentityConfig.store.arcgis_mock_fallback
        ArcgisApi::Mock::Geocoder.new
      else
        ArcgisApi::Geocoder.new
      end
    end
  end
```

so callsites would look like:

```ruby
ArcgisApi::GeocoderRouter.geocoder
```

Reply:
Everything looks good in here. I like this suggestion.

Reply:
This is a factory pattern, so using `Factory` in the name makes more sense and is more informative to developers who aren't as familiar with this specific codebase. The `Router` naming is confusing because that is normally associated with routing web requests to a controller.

Reply:
I thought about making this a module, but based on my prior experience using an instance tends to be more useful and reusable. The decision does seem a little arbitrary without more established conventions around IoC though.